### PR TITLE
Telecrystal clothing nerf

### PR DESCRIPTION
**Pull request by flappybatpal**
## About the PR 
[BALANCE][INPUT WANTED]

## Why's this needed? 
Reduces the chance of teleporting someone else with a telecrystal material from 50% to 33%.
Reduces the chance of being teleported when wearing a telecrystal item and being attacked from 50% to 33% with a 25% chance of dropping the item!

## Changelog

```changelog
(u)Flappybat
(+)Telecrystal materials are less likely to warp when used on yourself/others. Self teleporting due to attacks has a 25% chance of dropping the item at the location you warped from.
```

### CODE REVIEW

**Comment by frawhst** [comment]
CodeDude steals the credit yet again

**Comment by Tarmunora** [comment]
I feel like it would be more amusing if was instead a chance for the item to teleport right out of the user's hands

**Comment by flappybatpal** [comment]
There was a suggestion for that, I'm not sure what is preferable? Would they become too unusable or should this just be more of a gimmick anyway? They are quite easy to make with an arc plater if you can get a TC which also aren't that rare.

**Comment by flappybatpal** [comment]
Changed the damage to 25% chance of losing the gloves.

File: code/modules/materials/Mat_MaterialProcs.dm [hunk 1]
```diff
@@ -318,19 +318,24 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
...
 
 /datum/materialProc/telecrystal_onattack
 	execute(var/obj/item/owner, var/mob/attacker, var/mob/attacked)
 		var/turf/T = get_turf(attacked)
-		if(prob(50))
+		if(prob(33))
 			if(istype(attacked) && !isrestrictedz(T.z)) // Haine fix for undefined proc or verb /turf/simulated/floor/set loc()
 				. = get_offset_target_turf(get_turf(attacked), rand(-8, 8), rand(-8, 8))
+				var/fail_msg = ""
+				if (prob(25))
```
Comment (Tarmunora) [inline code/modules/materials/Mat_MaterialProcs.dm:333]:
```suggestion
				if (prob(25) && attacker == attacked)
```

File: code/modules/materials/Mat_MaterialProcs.dm [hunk 2]
```diff
@@ -318,19 +318,24 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
...
+					fail_msg = " but you lose [owner]!"
+					attacker.drop_item(owner)
+					playsound(attacker.loc, "sound/impact_sounds/Flesh_Tear_3.ogg", 40)
```
Comment (Tarmunora) [inline code/modules/materials/Mat_MaterialProcs.dm:336]: reconsider the sound effect perhaps?

Comment (flappybatpal) [reply code/modules/materials/Mat_MaterialProcs.dm]: Changed to "Poof". Also added sounds for the other warps which were mysteriously silent, will help make it more obvious "Oh I got warped".

File: code/modules/materials/Mat_Materials.dm [hunk 3]
```diff
@@ -120,6 +120,7 @@
 	proc/addTrigger(var/list/L, var/datum/materialProc/D)
 		for(var/datum/materialProc/P in L)
 			if(P.type == D.type) return 0
+		D.owner = src
```
Comment (Tarmunora) [inline code/modules/materials/Mat_Materials.dm:123]: ?

Comment (flappybatpal) [reply code/modules/materials/Mat_Materials.dm:123]: Fix for Mat_Materials.dm not setting the src.owner, fix was given to me by Pali.

Comment (Tarmunora) [reply code/modules/materials/Mat_Materials.dm:123]: 🤷

**Pull request merged**